test_convert_time_columns fails due to a pandas bug

`tracetools_analysis.test_data_model_util.TestDataModelUtil test_convert_time_columns` currently fails on Noble due to a bug in pandas: https://github.com/pandas-dev/pandas/issues/55730

```
FAILED test/tracetools_analysis/test_data_model_util.py::TestDataModelUtil::test_convert_time_columns - AttributeError: 'numpy.dtypes.DateTime64DType' object has no attribute 'unit'
```

It was fixed: https://github.com/pandas-dev/pandas/pull/55812

However, the fix is only available in version >=2.2.0, and we're stuck with version 2.1.4 on Noble using apt: https://packages.ubuntu.com/noble/python3-pandas.

--- a/tracetools_analysis/test/tracetools_analysis/test_data_model_util.py
+++ b/tracetools_analysis/test/tracetools_analysis/test_data_model_util.py
@@ -17,6 +17,8 @@ from datetime import timezone
 from typing import Dict
 import unittest
 
+from packaging.version import Version
+from pandas import __version__ as pandas_version
 from pandas import DataFrame
 from pandas.testing import assert_frame_equal
 
@@ -34,6 +36,10 @@ class TestDataModelUtil(unittest.TestCase):
             *args,
         )
 
+    @unittest.skipIf(
+        Version(pandas_version) < Version('2.2.0'),
+        'skip due to missing fix: pandas-dev/pandas#55812',
+    )
     def test_convert_time_columns(self) -> None:
         input_df = DataFrame(
             data=[

